backend/components/installer/src/test/java/co/airy/core/api/components/installer/CatalogHandlerTest.java

```diff
@@ -89,88 +82,57 @@ static void afterAll() throws Exception {
         kafkaTestHelper.afterAll();
     }
 
-
-    @Test
-    public void canOnApplicationEvent(@TempDir File tempDir) throws Exception {
-        callOnApplicationEvent(tempDir);
-    }
-
     @Test
     public void canGetComponents(@TempDir File tempDir) throws Exception {
         callOnApplicationEvent(tempDir);
 
-        final V1Job job = new V1Job()
-            .metadata(new V1ObjectMeta().name("helm-installed").namespace("test-namespace"));
-
-        //TODO: Move all of this to InstalledComponentsHandlerTest class & mock here
-        doReturn(job).when(helmJobHandler).launchHelmJob(eq(job.getMetadata().getName()), cmd.capture());
-        doReturn("helm-installed-test").when(helmJobHandler).waitForCompletedStatus(isA(CoreV1Api.class), eq(job));
-
-        final MockedConstruction.MockInitializer<CoreV1Api> fn = (mock, context) -> {
-            final ApiResponse<String> response = new ApiResponse<>(
-                    200,
-                    null,
-                    getInstalledComponents());
-
-            doReturn(response).when(mock).readNamespacedPodLogWithHttpInfo(
-                "helm-installed-test",
-                job.getMetadata().getNamespace(),
-                "",
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null);
-        };
-
-        try (MockedConstruction<CoreV1Api> apiMock = Mockito.mockConstruction(CoreV1Api.class, fn)) {
-            List<ComponentDetails> listComponents = catalogHandler.listComponents();
-
-            assertThat(cmd.getValue().size(), equalTo(3));
-            assertThat(cmd.getValue().get(2), equalTo("helm -n test-namespace list | awk '{print $1}' | tail -n +2")); 
-
-            //NOTE: We are just going to get some of the components in the list, and check his validity
-            ComponentDetails enterpriseSalesforceContactsIngestion = listComponents
-                .stream()
-                .filter((c -> c.getName().equals("enterprise-salesforce-contacts-ingestion")))
-                .findAny()
-                .orElse(null);
-            assertThat(enterpriseSalesforceContactsIngestion, is(notNullValue()));
-            assertThat(enterpriseSalesforceContactsIngestion.getInstallationStatus(), equalTo(InstallationStatus.installed));
-
-            ComponentDetails cognigyConnector = listComponents
-                .stream()
-                .filter((c -> c.getName().equals("cognigy-connector")))
-                .findAny()
-                .orElse(null);
-            assertThat(cognigyConnector, is(notNullValue()));
-            assertThat(cognigyConnector.getInstallationStatus(), equalTo(InstallationStatus.uninstalled));
-        }
-
+        doReturn(getInstallationStatues()).when(installationStatusComponentsHandler).getInstallationStatusComponentsCache();
+
+        final List<ComponentDetails> listComponents = catalogHandler.listComponents();
```

Comment:
It would be great if we could mock this. Otherwise, this test might fail again if the data in the catalog changes. What do you think?

Reply:
Well this is an integration test. and I think the test should fail. If for example, someone does something wrong on the `catalog` page. Otherwise we have no way to know something went wrong until is in production. I will suggest to mock it when we have CI/Test for the catalog repo.